src/core/middlewares/regquota/handler.go

```diff
@@ -0,0 +1,68 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package regquota
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/docker/distribution"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"github.com/goharbor/harbor/src/core/middlewares/util"
+	"io/ioutil"
+	"net/http"
+)
+
+type regQuotaHandler struct {
+	next http.Handler
+}
+
+// New ...
+func New(next http.Handler) http.Handler {
+	return &regQuotaHandler{
+		next: next,
+	}
+}
+
+// ServeHTTP PATCH manifest ...
+func (rqh regQuotaHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	match, _, _ := util.MatchManifestURL(req)
+	if match {
+		var mfSize int64
+		var mfDigest string
+		mediaType := req.Header.Get("Content-Type")
+		if req.Method == http.MethodPut && mediaType == "application/vnd.docker.distribution.manifest.v2+json" {
```

Comment:
There are some other media types that should be supported, refer to  https://github.com/goharbor/harbor/blob/master/src/replication/transfer/image/transfer.go#L263